### PR TITLE
Rewrite config form the Security bundle to use ConfigBuilders

### PR DESCRIPTION
**Pull request by Nyholm**

### CODE REVIEW

File: security/custom_authentication_provider.rst [hunk 1]
```diff
@@ -634,19 +634,18 @@ set to any desirable value per firewall.
...
-                        'lifetime' => 30,
-                    ],
-                ],
-            ],
-        ]);
+            $security->firewall('wsse_secured')
+                ->pattern('^/api/')
+                ->stateless(true)
+                ->wsse(true)
+                    ->lifetime(30)
```
Comment (Nyholm) [inline security/custom_authentication_provider.rst]: I need to confirm that this is working

File: security.rst [hunk 2]
```diff
@@ -548,29 +545,29 @@ You must enable this using the ``login_throttling`` setting:
...
+            // ...
+            $mainFirewall = $security->firewall('main');
 
-                    // configure the maximum login attempts (per minute)
-                    'login_throttling' => [
-                        'max_attempts' => 3,
-                    ],
+            // by default, the feature allows 5 login attempts per minute
+            $mainFirewall
+                ->loginThrottling(null);
```
Comment (Nyholm) [inline security.rst]: Im not sure this is correct. I need to confirm and possibly make a PR to symfony/symfony to fix int

File: security/custom_authentication_provider.rst (hunk 1 above)
Comment (Nyholm) [reply security/custom_authentication_provider.rst]: It is working!

File: security.rst (hunk 2 above)
Comment (Nyholm) [reply security.rst]: I misunderstood this config fist. I though this config would disable login throttling. 

```yaml
security:
    enable_authenticator_manager: true

    firewalls:
        main:
            pattern: /
            login_throttling: null
```

I was wrong. It actually enables it.. The PR is correct.

**Comment by Nyholm** [comment]
Status: needs review

**Comment by javiereguiluz** [comment]
Tobias, you did a great work here. Thanks a lot 🤗

**Pull request merged**